libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -102,7 +107,7 @@ instance
           -- TODO, we should make sure size' is greater than or equal to 0
           satisfies
             (sz + Lit (sizeOf mustMap))
-            (size <> cardinality (mapSpec fstFn $ mapSpec toGenericFn kvs))
+            (maybe TrueSpec leqSpec mHint <> size <> cardinality (mapSpec fstFn $ mapSpec toGenericFn kvs))
```

Comment:
This is where the rubber hits the road.  Maybe a comment about how the hint is used?
I guess this relates to the use of 'min' I talked about earlier.